src/cc/frontends/p4/compiler/ebpfAction.py

```diff
@@ -0,0 +1,327 @@
+            raise CompilationException(True, "Unexpected primitive action {0}", actionName)
+
+    def serializeCount(self, caller, arguments, serializer, dataContainer, program):
+        assert isinstance(serializer, ProgramSerializer)
+        assert isinstance(program, ebpfProgram.EbpfProgram)
+        assert isinstance(arguments, list)
+        assert len(arguments) == 2
+
+        counter = arguments[0]
+        index = ArgInfo(arguments[1], caller, dataContainer, program)
+        ctr = program.getCounter(counter.name)
+        assert isinstance(ctr, ebpfCounter.EbpfCounter)
+        serializer.emitIndent()
+        serializer.blockStart()
+
+        # This is actually incorrect, since the key is not always an u32.
+        # This code is currently disabled
+        key = "ebpf_index"
+        serializer.emitIndent()
+        serializer.appendFormat("u32 {0} = {1};", key, index.asString)
+        serializer.newline()
+
+        ctr.serializeCode(key, serializer, program)
+
+        serializer.blockEnd(True)
+
+    def serializeCallee(self, caller, callee, arguments, serializer, dataContainer, program):
+        if self.invalid:
+            raise CompilationException(True, "{0} Attempting to generate code for an invalid action", self.hliraction)
+
+        assert isinstance(serializer, ProgramSerializer)
+        assert isinstance(program, ebpfProgram.EbpfProgram)
+        assert isinstance(callee, p4_action)
+        assert isinstance(arguments, list)
+
+        if callee.name in EbpfAction.unsupported:
+            raise NotSupportedException("{0}", callee)
+
+        # This is not yet ready
+        #if callee.name == "count":
+        #    self.serializeCount(caller, arguments, serializer, dataContainer, program)
+        #    return
+
+        serializer.emitIndent()
+        args = self.transformArguments(arguments, caller, dataContainer, program)
+        if callee.name == "modify_field":
+            dst = args[0]
+            src = args[1]
+
+            size = self.checkSize(callee, [a.widthInBits() for a in args], program)
+            if size is None:
+                raise CompilationException(True, "Cannot infer width for arguments {0}", callee)
+            elif size <= 32:
+                serializer.appendFormat("{0} = {1};",
+                                        dst.asString,
+                                        src.asString)
+            else:
+                if not dst.isLvalue:
+                    raise NotSupportedException("Constants wider than 32-bit: {0}({1})", dst.caller, dst.asString)
+                if not src.isLvalue:
+                    raise NotSupportedException("Constants wider than 32-bit: {0}({1})", src.caller, src.asString)
+                serializer.appendFormat("memcpy(&{0}, &{1}, {2});",
+                                        dst.asString,
+                                        src.asString,
+                                        size / 8)
+        elif callee.name == "add" or \
```

Comment:
The more canonical way to do multi-line if statements is by wrapping with () and omitting the \ on each line.